simulator.py

```diff
@@ -59,8 +59,8 @@ def schedule_prediction(self, delay, prediction):
         self.new_prediction(prediction)
 
     def kill_producers(self):
-        self.logger.info("Killing producers ...")
         self.manager.terminate_producers()
+        self.manager.terminate_consumers()
```

Comment:
This seems to fix the problem with haning threads blocking the simulator from terminating. Should be investigated more, but I imagine this will be fixed when we have complete optimization accept/reject logic.

Reply:
It might be because the consumers are sending a ask() and waiting for a reply? ask() has a blocking wait.